stdlib/public/Concurrency/TaskGroup.swift

```diff
@@ -104,60 +97,54 @@ public func withTaskGroup<ChildTaskResult: Sendable, GroupResult>(
-///       // some accumulation logic (e.g. sum += result)
+///     while let result = await group.next() {
+///         // some accumulation logic (e.g. sum += result)
 ///     }
 ///
 /// It is also possible to collect results from the group by using its
 /// `AsyncSequence` conformance, which enables its use in an asynchronous for-loop,
 /// like this:
 ///
-///     for try await result in group {
-///       // some accumulation logic (e.g. sum += result)
+///     for await result in group {
+///         // some accumulation logic (e.g. sum += result)
 ///      }
+/// ◊TR: Which of the above do we prefer?  Why give two recommendations?
+///
+/// When tasks are added to the group
+/// using the `Group.spawn(priority:operation:)` method,
+/// they might begin executing immediately.
+/// Even if their results are not collected explicitly and such task throws,
+/// and was not yet canceled,
+/// it may result in the `withTaskGroup` throwing.
+/// ◊TR: What does the above "such task throws" mean?
 ///
-/// ### Thrown errors
-/// When tasks are added to the group using the `group.spawn` function, they may
-/// immediately begin executing. Even if their results are not collected explicitly
-/// and such task throws, and was not yet cancelled, it may result in the `withTaskGroup`
-/// throwing.
+/// Task Group Cancellation
+/// =======================
 ///
-/// ### Cancellation
-/// If the task that the group is running in is cancelled, the group becomes 
-/// cancelled and all child tasks spawned in the group are cancelled as well.
-/// 
-/// If an error is thrown out of the task group, all of its remaining tasks
-/// will be cancelled and the `withTaskGroup` call will rethrow that error.
+/// Canceling the task in which the group is running
+/// also cancels the group and all of its child tasks.
 ///
-/// Individual tasks throwing results in their corresponding `try group.next()`
-/// call throwing, giving a chance to handle individual errors or letting the
-/// error be rethrown by the group.
+/// If an error is thrown by one of the tasks in a task group,
+/// all of its remaining tasks are canceled,
+/// and the `withTaskGroup` method rethrows that error.
 ///
-/// Postcondition:
-/// Once `withThrowingTaskGroup` returns it is guaranteed that the `group` is *empty*.
+/// An individual task throws its error
+/// in the corresponding call to `Group.next()`,
+/// which gives you a chance to handle individual error
+/// or to let the error be rethrown by the group.
 ///
-/// This is achieved in the following way:
-/// - if the body returns normally:
-///   - the group will await any not yet complete tasks,
-///     - if any of those tasks throws, the remaining tasks will be cancelled,
-///   - once the `withTaskGroup` returns the group is guaranteed to be empty.
-/// - if the body throws:
-///   - all tasks remaining in the group will be automatically cancelled.
+/// After this method returns, the task group is guaranteed to be empty.
```

Comment:
👍 

Maybe worth a `### Postconditions`?

It kind of sneaks into the cancellation section now which isn't quite right

Reply:
Good catch that this doesn't belong under the cancellation section.  I'd like to avoid such a short section, so I've moved this up to be before the examples above.